File: NodeJS/esercizio5.js
#!/usr/bin/env node
/*
Esercizio: scrivere una funzione che calcoli il fattoriale di un numero intero
non negativo
*/

'use strict';

var readline = require('readline-sync');

var input = parseInt(readline.question("Inserisci un numero intero (piccolo!): "));

// Calcola il fattoriale
var output = 1;
for (var i = input; i > 0; i--) {
	output *= i;
}
// Stampo il risultato
console.log("Il fattoriale di " + input + " è " + output + ".");

File: C/esercizio5.c
/*
Esercizio: scrivere una funzione che calcoli il fattoriale di un numero intero
non negativo
*/

#include <stdio.h>
#include <stdlib.h>

int main() {
	int input;
	
	// Richiedi il numero
	printf("Inserisci un numero intero (piccolo!): ");
	scanf("%d", &input);

	// Calcola il fattoriale
	int output = 1;
	for (int i = input; i > 0; i--) {
		output *= i;
	}

	// Stampo il risultato
	printf("Il fattoriale di %d è %d.\n", input, output);

	return 0;
}

